src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import static java.lang.Math.abs;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int RESIZE_FACTOR = 2;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        putInSize(key, value);
+    }
 
+    private void putInSize(K key, V value) {
+        int bucketNumber = getBucketNumber(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        Node<K, V> currentNode = table[bucketNumber];
+        if (table[bucketNumber] == null) {
+            table[bucketNumber] = newNode;
+            size++;
```

Comment:
put size++; at the end of the method, it's common for if-else clauses, and when we rewrite exiting method we call return so we won't reach it